scripts/buildipxe

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/bash
+
+set -euxo pipefail
+# ENV VARIABLE CONFIG #
+# shellcheck disable=SC1091
+. /bin/tls-common.sh
+. /bin/ironic-common.sh
+
+# This script should be used as an init container's entry point thus
+# it is expected that the user would like to build the ipxe when this script
+# is started. By default the script will try to build from cache and fails
+# if the cache is unavailable.
+export IPXE_BUILD_FROM_CACHE="${IPXE_BUILD_FROM_CACHE:-true}"
+export IPXE_BUILD_FROM_REPO="${IPXE_BUILD_FROM_REPO:-false}"
+export IPXE_SHARED_FIRMWARE_SOURCE="${IPXE_SHARED_FIRMWARE_SOURCE:-/shared/ipxe-source}"
+export IPXE_EMBED_SCRIPT="${IPXE_EMBED_SCRIPT:-/bin/embed.ipxe}"
+export IPXE_EMBED_SCRIPT_TEMPLATE="${IPXE_EMBED_SCRIPT_TEMPLATE:-/bin/embed.ipxe.j2}"
+export IPXE_RELEASE_BRANCH="${IPXE_RELEASE_BRANCH:-v1.21.1}"
+export IPXE_ENABLE_IPV6="${IPXE_ENABLE_IPV6:-false}"
+export IPXE_ENABLE_HTTPS="${IPXE_ENABLE_TLS:-false}"
+# BUILD_OPTIONS are not configurable directly
+export IPXE_BUILD_OPTIONS="NO_WERROR=1 EMBED=${IPXE_EMBED_SCRIPT}"
+# PREPARE SOURCE #
+
+# Create debug folder
+mkdir -p "/shared/ipxe-debug"
+
+# In case building ipxe firmware from shared volume
+if [[ "${IPXE_BUILD_FROM_CACHE}" == "true" ]]; then
+    if [[ -r "${IPXE_SHARED_FIRMWARE_SOURCE}" ]]; then
+        cp -r "${IPXE_SHARED_FIRMWARE_SOURCE}" "/tmp"
+        ls -all "/tmp/ipxe-source"
+    else
+        >&2 echo "ERROR: can't build ipxe from cache, there is no path!"
+        exit 1
+    fi
+# In case building ipxe firmware from upstream git repo directly
+# Requires Internet access!
+elif [[ "${IPXE_BUILD_FROM_CACHE}" == "false" ]] \
+         && [[ "${IPXE_BUILD_FROM_REPO}" == "true" ]]; then
+    git clone --depth 1 --branch "${IPXE_RELEASE_BRANCH}" \
+        "https://github.com/ipxe/ipxe.git" \
+        "/tmp/ipxe-source"
+fi
```

Comment:
I wonder if it would make sense to validate the variables before this? My understanding is that we require either `IPXE_BUILD_FROM_CACHE=true && IPXE_BUILD_FROM_REPO=false` or the other way around so we could check that.

Alternatively, we can make this a bit easier on the user by doing more like a fall through. If user asks to build from cache, then try it. If it fails fall back to repo. If we go with this alternative I would say we do not need an explicit variable for `IPXE_BUILD_FROM_REPO`.

Reply:
I see your point, but my logic here was to make cache based building take precedence, and I would like to avoid fall through logic as much as possible, my idea was to have explicit config with explicit precedence and to fail hard in case pxe buildig was needed but not configured.

Reply:
And the reason that cache based building takes precedence is that I would like to avoid reaching for external git repos a secondary option, I consider local cache a safer default choice.

Reply:
Ok I'm fine with no fallthrough. However, I still think some validation could be helpful here instead of ending up below at 

> ERROR: The ipxe firmware source is missing, check the env vars!

It may not be obvious what variables to check.

Reply:
Do I understand correctly, that you are proposing an extra check that would check both IPXE_BUILD_FROM_CACHE and IPXE_BUILD_FROM_REPO and if both of these are false it would say that these are and the build can't be executed ? That is fine by me  I can add that, I agree that checking /tmp/ipxe-source is not the best approach.

Reply:
Exactly, it would be helpful to have this check I think

Reply:
done.